src/box/txn.c

```diff
@@ -303,7 +313,7 @@ txn_begin(void)
 	txn->fiber = NULL;
 	fiber_set_txn(fiber(), txn);
 	/* fiber_on_yield is initialized by engine on demand */
-	trigger_create(&txn->fiber_on_stop, txn_on_stop, NULL, NULL);
+	trigger_create(&txn->fiber_on_stop, txn_on_stop, txn, NULL);
```

Comment:
All time when we finish to process iproto message, we  call `fiber_set_txn(fiber(), NULL);`. If trigger will be triggered (fiber_on_yeild) (in case when  memtx_use_mvcc_engine == false it's crashed) For txn_on_stop it's done for symmetry.